app/api/ada/lib/storage/bridge/lovefieldDerive.js

```diff
@@ -0,0 +1,122 @@
+// @flow
+
+import type {
+  lf$Database,
+  lf$Transaction,
+} from 'lovefield';
+
+import { IDerive } from '../models/functionalities/IDerive';
+
+import { getAllSchemaTables, raii, } from '../database/utils';
+import {
+  AddPublicDeriver,
+  DerivePublicFromPrivate,
+} from '../database/genericBip44/api/add';
+import type {
+  DerivePublicFromPrivateRequest
+} from '../database/genericBip44/api/add';
+import type { KeyRow } from '../database/uncategorized/tables';
+
+import {
+  decryptKey,
+  deriveKey,
+  toKeyInsert
+} from './utils';
+import type {
+  KeyInfo,
+} from './utils';
+
+export type LovefieldDeriveRequest = {|
+  ...DerivePublicFromPrivateRequest<{}>,
+  decryptPrivateDeriverPassword: ?string,
+  publicDeriverPublicKey?: KeyInfo,
+  publicDeriverPrivateKey?: KeyInfo,
+|};
+export async function derivePublicDeriver(
```

Comment:
add comment saying we don't lock here but lock in the _derive to help hook to walletbuilder